solutions/balancing_parentheses.py

```diff
@@ -0,0 +1,122 @@
+def balancing_parentheses(string):
+
+    # solve keeping dynamic track of parenthesis mismatch
+    openings = 0
+    missings = 0
+
+    for i in range(len(string)):
+
+        if string[i] == '(':
+            openings += 1
+            continue
+
+        if openings > 0:
+            openings -= 1
+        else:
+            missings += 1
+
+    return openings + missings
+
+    # solve using stack data structure
+    '''
+    stack = []
+
+    extra = 0
+
+    for i in range(len(string)):
+
+        if string[i] == '(':
+
+            stack.insert(0, '(')
```

Comment:
The `insert` method is adding elements from the left side of the list and the `pop` method is taking it out from the right side. The stack should add and remove from the same side.

Suggestions:

- Use the `append` and `pop` method if you'd like to model a stack.
- Use the `appendleft` and `pop` methods of the built-in [deque collection](https://docs.python.org/3/library/collections.html#collections.deque) if you'd like to model a single ended queue.